tx-parser/src/lib.rs

```diff
@@ -30,181 +34,128 @@ fn fn_parser<O, I>(f: fn(I) -> ParseResult<O, I>, err: &'static str) -> TxParser
     parser(f).expected(err)
 }
 
-impl<I> Tx<I>
-    where I: Stream<Item = Value>
-{
-    fn integer() -> TxParser<i64, I> {
-        fn_parser(Tx::<I>::integer_, "integer")
-    }
+def_value_satisfy_parser_fn!(Tx, integer, i64, Value::as_integer);
 
-    fn integer_(input: I) -> ParseResult<i64, I> {
-        return satisfy_map(|x: Value| if let Value::Integer(y) = x {
-                Some(y)
+fn value_to_namespaced_keyword(val: &Value) -> Option<NamespacedKeyword> {
+    if let &Value::NamespacedKeyword(ref i) = val { Some(i.clone()) } else { None }
```

Comment:
Sigh. Nope. I'll do it your way and land it, and work out the fancy thing later.